core/lib/mcxtx.h

```diff
@@ -245,5 +245,5 @@
      * DrMi#665: we now preserve all of the xmm registers.
      */
 #    endif
-    dr_ymm_t ymm[MCXT_NUM_SIMD_SLOTS];
+    dr_zmm_t simd[MCXT_NUM_SIMD_SLOTS];
```

Comment:
Please add a (non-exported) comment talking about size being a potential issue for stack buffers or in general and how we may want to separate this out to the heap so we can A) save space on memory-constrained platforms (which prob don't support the latest SIMD) and B) keep our signal stack size small.

Reply:
Ok. Regarding B), actually not sure yet about this. The Linux kernel might do something similar to DynamoRIO on a system level AFAIK. e.g. http://events17.linuxfoundation.org/sites/events/files/slides/LinuxCon_NA_2014.pdf
Basically doing some lazy context saving based on whether AVX-512 apps had been seen.  So we might just measure the xstate's size and then copy only if the kernel does? But then I guess the xstate's size is statically compiled, so it would still be part of the signal's frame, so you're right that while we can avoid the copy, we can't avoid using the memory unless it's a pointer.

Reply:
Done.